README.md

```diff
@@ -502,6 +503,43 @@ class MyGem::Inflector < Zeitwerk::GemInflector
 end
 ```
 
+<a id="markdown-the-on_load-callback" name="the-on_load-callback"></a>
+### The `on_load` callback
+
+The usual place to run something when a file is loaded is the file itself. However, sometimes you'd like to be called, and this is possible with the `on_load` callback.
+
+For example, let's say a class managed by Zeitwerk in a web application has a class-level attribute whose value depends on the environment:
+
+```ruby
+class SomeApiClient
+  class << self
+    attr_accessor :endpoint
+  end
+end
+```
+
+With `on_load`, it is easy to initialize said attribute at boot time according to the configuration:
+
+```ruby
+# config/environments/development.rb
+loader.on_load("SomeApiClient") do
+  SomeApiClient.endpoint = "https://api.dev"
+end
+
+# config/environments/production.rb
+config.on_load("SomeApiClient") do
```

Comment:
It should be `loader` here right? Also in the changelog.

Reply:
Good catch, it was introduced to see if reviewers paid attention 🤣.

Reply:
I have to admit I never review docs cause I'm so bad at it...